lib/elixir/lib/kernel/typespec.ex

```diff
@@ -200,25 +199,17 @@ defmodule Kernel.Typespec do
   ## Translation from Elixir AST to typespec AST
 
   @doc false
-  def translate_typespecs_for_module(set, bag) do
+  def translate_typespecs_for_module(_set, bag) do
     types = Enum.map(take_typespec(bag, :type), &translate_type/1)
     specs = Enum.map(take_typespec(bag, :spec), &translate_spec/1)
     callbacks = Enum.map(take_typespec(bag, :callback), &translate_spec/1)
     macrocallbacks = Enum.map(take_typespec(bag, :macrocallback), &translate_spec/1)
-    optional_callbacks = List.flatten(take_optional(set, :optional_callbacks))
+    optional_callbacks = List.flatten(get_typespec(bag, {:accumulate, :optional_callbacks}))
```

Comment:
Is there a reason why we delete some attributes from the table, but keep `optional_callbacks`?

Reply:
Because the other ones are internal but `@optional_callbacks` can be accessed by the user in a `after_compile` callback, so we can't remove it.

Reply:
I was more asking, if we need to delete the others - if not, we could just read them, they would be removed with the whole table.

Reply:
Could you see a benefit for not removing? I thought I would release the memory available early and deletes are not expensive (unless they generate more garbage but unlikely?).

Reply:
Deleting attributes might resize the table and needs to grab a write lock instead of just the read lock. I'm not sure that would be important in that case, since there's just one process accessing the table anyway.